src/index.ts

```diff
@@ -305,37 +305,101 @@ export type StartEmulatorOptions = {
-          /** Whether the emulator should be created and managed by cyanoacrylate. */
-          managed?: false;
-      }
-) & {
+/** Run target options for an Android emulator that the user creates, manages, and starts themselves. */
+export type AndroidEmulatorRunTargetOptionsUnmanagedSelfStarted = {
+    startEmulatorOptions?: undefined;
+    /** The name of a snapshot to use when resetting the emulator. */
+    snapshotName?: string;
+
+    /** Whether the emulator should be created and managed by cyanoacrylate. */
+    managed?: false;
+};
+/**
+ * Run target options for an Android emulator that the user creates and manages themselves but that is automatically
+ * started by cyanoacrylate.
+ */
+export type AndroidEmulatorRunTargetOptionsUnmanagedStarted = {
+    /** Options to configure how the emulator should be started by cyanoacrylate. */
+    startEmulatorOptions: {
+        /** The name of the emulator to start. */
+        emulatorName: string;
+    } & StartEmulatorOptions;
+
     /** The name of a snapshot to use when resetting the emulator. */
     snapshotName?: string;
-    /** Options for the emulator if you want it to be automatically managed by this library. */
+
+    /** Whether the emulator should be created and managed by cyanoacrylate. */
+    managed?: false;
+};
+/** Run target options for an Android emulator that is created, managed, and started automatically by cyanoacrylate. */
+export type AndroidEmulatorRunTargetOptionsManaged = {
+    /** Options to configure how the emulator should be created by cyanoacrylate. */
+    managedEmulatorOptions: {
+        /**
+         * A key to distinguish the emulator from other ones created by cyanoacrylate. All analyses using the same key
+         * will share an emulator. The created emulator will be named `cyanoacrylate-{key}-{MD5 hash of the options}`.
+         */
+        key: string;
+
+        /** Which kind of emulator to create. */
+        createEmulatorOptions?: EmulatorOptions;
+
+        // /** @unimplemented Optional honey data to place on the device before creating the snapshot. */
+        // honeyData?: {
+        //     /** The device name to set, which shows up to other network or bluetooth devices. */
+        //     deviceName?: string;
+        //     /** Content to put into the clipboard. */
+        //     clipboard?: string;
+        //     /** Calendar events to place on the device. */
+        //     calendarEvents?: CalendarEventData[];
+        //     /** Contacts to place on the device. */
+        //     contacts?: ContactData[];
+        // };
+
+        /**
+         * How often to try rebuilding the emulator completely in case of an error.
+         *
+         * @default 0
+         */
+        attemptRebuilds?: number;
+    };
+
+    /** Whether the emulator should be created and managed by cyanoacrylate. */
+    managed: true;
+
+    /** The name of a snapshot to use when resetting the emulator. */
+    snapshotName?: string;
```

Comment:
How would this work? The user would have to start an analysis to have us create the emulator, start the emulator, manually create a snapshot, stop the analysis, and then start a new analysis with different parameters (the name of the snapshot the created).

I really don't see how managed emulators can work without snapshot management.